packages/overlay/src/OverlayBase.ts

```diff
@@ -0,0 +1,908 @@
+                    composed: true,
+                });
+                event.relatedTarget.addEventListener(
+                    relationEvent.type,
+                    (event: Event) => {
+                        if (!event.composedPath().includes(this)) {
+                            this.open = false;
+                        }
+                    }
+                );
+                event.relatedTarget.dispatchEvent(relationEvent);
+            });
+        }
+        if (!this.hasAttribute('id')) {
+            this.setAttribute(
+                'id',
+                `${this.tagName.toLowerCase()}-${crypto
+                    .randomUUID()
+                    .slice(0, 8)}`
+            );
+        }
+        if (
+            changes.has('open') &&
+            (typeof changes.get('open') !== 'undefined' || this.open)
+        ) {
+            this.manageOpen(changes.get('open'));
+        }
+        if (changes.has('trigger')) {
+            const [id, interaction] = this.trigger?.split('@') || [];
+            this.elementResolver.selector = id ? `#${id}` : '';
+            this.triggerInteraction = interaction as
+                | 'click'
+                | 'longpress'
+                | 'hover'
+                | undefined;
+        }
+        const oldTrigger = this.triggerElement as HTMLElement;
+        if (changes.has(elementResolverUpdatedSymbol)) {
+            this.triggerElement = this.elementResolver.element;
+            this.manageTriggerElement(oldTrigger);
+        }
+        if (changes.has('triggerElement')) {
+            this.manageTriggerElement(changes.get('triggerElement'));
+        }
+    }
+
+    protected override updated(changes: PropertyValues): void {
+        super.updated(changes);
+        if (changes.has('placement')) {
+            if (this.placement) {
+                this.dialogEl.setAttribute('actual-placement', this.placement);
+            } else {
+                this.dialogEl.removeAttribute('actual-placement');
+            }
+            if (this.open && typeof changes.get('placement') !== 'undefined') {
+                this.placementController.resetOverlayPosition();
+            }
+        }
+    }
+
+    protected renderContent(): TemplateResult {
+        return html`
+            <div part="content">
+                <slot @slotchange=${this.handleSlotchange}></slot>
+            </div>
+        `;
+    }
+
+    protected renderDialog(): TemplateResult {
+        return html`
+            <dialog
+                class="dialog"
+                part="dialog"
+                @close=${this.handleBrowserClose}
+                @cancel=${this.handleBrowserClose}
+                @beforetoggle=${this.handleBeforetoggle}
+                style=${styleMap({
+                    '--swc-overlay-open-count':
+                        OverlayBase.openCount.toString(),
+                    translate: this.hasUpdated ? null : '-999em -999em',
+                })}
```

Comment:
As long as this is only ever computed once, at first render, I get how it will capture the global `openCount` when it's opened, and use the correct z-index.

But if `renderDialog` is recomputed later, for example after `openCount` has incremented, would it not update this value (& this its z-index) to be at the wrong level?

Reply:
The intent here is for the last opened Overlay to always have the highest value as a corollary to top layer being a single stack in fallback browsers. I'll add a comment.